torch/distributions/multinomial.py

```diff
@@ -70,9 +70,13 @@ def expand(self, batch_shape, _instance=None):
     def _new(self, *args, **kwargs):
         return self._categorical._new(*args, **kwargs)
 
-    @constraints.dependent_property
+    @constraints.dependent_property(is_discrete=True, event_dim=1)
     def support(self):
-        return constraints.integer_interval(0, self.total_count)
+        total_count = self.total_count
+        if hasattr(total_count, "unsqueeze"):
+            total_count = total_count.unsqueeze(-1)
+        return constraints.independent(
+            constraints.integer_interval(0, total_count), 1)
```

Comment:
I think it is better to have `constraints.multinomial`.

Reply:
Good idea. I've added a `constraints.multinomial` in this PR. However note that it checks the slightly weaker condition
```diff
- (0 <= x).all(-1) & (x.sum(-1) == total_count)
+ (0 <= x).all(-1) & (x.sum(-1) <= total_count)
```
because we allow passing a bogus `total_count.max()` to `Multinomial` when it is used as a likelihood. I plan to try to address this in the future while fixing #42407.